### ISSUE TITLE
Cannot read property name of null - inconsistent behavior

### ISSUE THREAD

**Issue by alechp** [sindresorhus/is-reachable#31]
TLDR: 
👍 - Both `localhost:4000` and `http://localhost:4000` work as expected. 
👎 - Only `http://localhost:4466` works. `localhost:4466` fails

## Case 1 - 👍
> with http://

**Input**

```js
test(`localhost:4000 is reachable`, async t => {
  t.true(await isReachable("http://localhost:4000"));
});

test(`localhost:4466 is reachable`, async t => {
  t.true(await isReachable("http://localhost:4466"));
});
```

**Output**
![screen shot 2018-09-11 at 3 43 18 pm](https://user-images.githubusercontent.com/3058939/45391856-847b3d80-b5d9-11e8-9c55-63650825907a.png)

## Case 2 - 👍 / 👎
> with and without http://

**Input**

```js
test(`localhost:4000 is reachable`, async t => {
  t.true(await isReachable("http://localhost:4000"));
});

test(`localhost:4466 is reachable`, async t => {
  t.true(await isReachable("http://localhost:4466"));
});

test(`localhost:4000 is reachable`, async t => {
  t.true(await isReachable("localhost:4000"));
});

test(`localhost:4466 is reachable`, async t => {
  t.true(await isReachable("localhost:4466"));
});
```

**Output**
![screen shot 2018-09-11 at 3 43 27 pm](https://user-images.githubusercontent.com/3058939/45391879-9e1c8500-b5d9-11e8-9708-f3bfe2c06547.png)

## Case 3 - 👍 / 👎
> without http://

**Input**
```js
test(`localhost:4000 is reachable`, async t => {
  t.true(await isReachable("localhost:4000"));
});

test(`localhost:4466 is reachable`, async t => {
  t.true(await isReachable("localhost:4466"));
});
```

**Output**
![screen shot 2018-09-11 at 3 58 46 pm](https://user-images.githubusercontent.com/3058939/45392305-9f4eb180-b5db-11e8-8e22-98b7c2be649f.png)

---

### Chrome Screenshots
> Included this so you can see both are **http://**

![screen shot 2018-09-11 at 3 47 16 pm](https://user-images.githubusercontent.com/3058939/45391969-0f5c3800-b5da-11e8-817d-b6c27e78b423.png)
![screen shot 2018-09-11 at 3 47 19 pm](https://user-images.githubusercontent.com/3058939/45391970-0f5c3800-b5da-11e8-86fc-e5a27c577c50.png)

**Comment by sindresorhus** [comment]
Really weird that there's a difference as we just prepend `http://` to protocol-less URLs, so they should be equivalent: https://github.com/sindresorhus/is-reachable/blob/d45c4da0c93683fef09ed9b3e46919c887cb85f5/index.js#L23

**Comment by moghthalkushal** [comment]
> TLDR:
> 👍 - Both `localhost:4000` and `http://localhost:4000` work as expected.
> 👎 - Only `localhost:4466` works. `http://localhost:4466` fails
> 
> ## Case 1 - 👍
> > with http://
> 
> **Input**
> 
> ```js
> test(`localhost:4000 is reachable`, async t => {
>   t.true(await isReachable("http://localhost:4000"));
> });
> 
> test(`localhost:4466 is reachable`, async t => {
>   t.true(await isReachable("http://localhost:4466"));
> });
> ```
> 
> **Output**
> ![screen shot 2018-09-11 at 3 43 18 pm](https://user-images.githubusercontent.com/3058939/45391856-847b3d80-b5d9-11e8-9c55-63650825907a.png)
> 
> ## Case 2 - 👍 / 👎
> > with and without http://
> 
> **Input**
> 
> ```js
> test(`localhost:4000 is reachable`, async t => {
>   t.true(await isReachable("http://localhost:4000"));
> });
> 
> test(`localhost:4466 is reachable`, async t => {
>   t.true(await isReachable("http://localhost:4466"));
> });
> 
> test(`localhost:4000 is reachable`, async t => {
>   t.true(await isReachable("localhost:4000"));
> });
> 
> test(`localhost:4466 is reachable`, async t => {
>   t.true(await isReachable("localhost:4466"));
> });
> ```
> 
> **Output**
> ![screen shot 2018-09-11 at 3 43 27 pm](https://user-images.githubusercontent.com/3058939/45391879-9e1c8500-b5d9-11e8-9708-f3bfe2c06547.png)
> 
> ## Case 3 - 👍 / 👎
> > without http://
> 
> **Input**
> 
> ```js
> test(`localhost:4000 is reachable`, async t => {
>   t.true(await isReachable("localhost:4000"));
> });
> 
> test(`localhost:4466 is reachable`, async t => {
>   t.true(await isReachable("localhost:4466"));
> });
> ```
> 
> **Output**
> ![screen shot 2018-09-11 at 3 58 46 pm](https://user-images.githubusercontent.com/3058939/45392305-9f4eb180-b5db-11e8-8e22-98b7c2be649f.png)
> 
> ### Chrome Screenshots
> > Included this so you can see both are **http://**
> 
> ![screen shot 2018-09-11 at 3 47 16 pm](https://user-images.githubusercontent.com/3058939/45391969-0f5c3800-b5da-11e8-817d-b6c27e78b423.png)
> ![screen shot 2018-09-11 at 3 47 19 pm](https://user-images.githubusercontent.com/3058939/45391970-0f5c3800-b5da-11e8-86fc-e5a27c577c50.png)

Yup True , i am facing same issues
